main/sceneManager.cpp

```diff
@@ -1,26 +1,66 @@
 #include "main/sceneManager.h"
 
-using Tri = objects::TriangleObj;
+SceneManager::SceneManager()
+{
+    createCollectors();
+}
+
+std::vector<objects::EnergyCollector *> SceneManager::getEnergyCollectors()
+{
+    std::vector<objects::EnergyCollector *> temp;
+    temp.reserve(_energyCollectors.size());
+    for (const auto &t : _energyCollectors)
+    {
+        temp.push_back(t.get());
+    }
+    return temp;
+}
+
+std::vector<objects::TriangleObj *> SceneManager::getModelTriangles()
+{
+    std::vector<objects::TriangleObj *> temp;
+    temp.reserve(_modelTriangles.size());
+    for (const auto &t : _modelTriangles)
+    {
+        temp.push_back(t.get());
+    }
+    return temp;
+}
 
-void IamSimulator()
+// PRIVATE METHODS
+
+void SceneManager::createCollectors()
 {
-    SceneManager manager;
-    CHECK(manager.loadModel("123231")) << "Cannot load model";
+    _energyCollectors.reserve(constants::kCollectors);
 
-    std::vector<Tri> triangles = manager.getModelTriangles();
-    generators::PointSource source;
+    const double collectorRadius = static_cast<double>(constants::kSimulationRadius) / 2;
+    double alphaIncrement;
 
-    for (int i = 0; i < 1000000; i++)
+    if (constants::kCollectors % 2 == 0)
```

Comment:
FME: Given number of collectors was a hard-coded constant why did you even added this line

Reply:
Because I plan to test progam with different numebers of kCollectors in the future. Maybe interface is wrong.

Reply:
Right, so if you want to change it, then this "constant" is not actually a constant.

Reply:
noted